### ISSUE TITLE
False positive for SC1003

### ISSUE THREAD

**Issue by ale5000-git** [koalaman/shellcheck#3032]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1003
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

echo 'abc test' | tr -d '\\'

```

#### Here's what shellcheck currently says:

echo 'abc test' | tr -d '\\'
                                    ^-- SC1003 (info): Want to escape a single quote? echo 'This is how it'\''s done'.

#### Here's what I wanted or expected to see:

It is the backslash that is escaped, not the single quote.

The backslash in tr need to be escaped, otherwise bash say this:
```
echo 'abc test' | tr -d '\'
tr: warning: an unescaped backslash at end of string is not portable
```

**Comment by wileyhy** [comment]
Perhaps the Exceptions section in the wiki could be altered.

https://www.shellcheck.net/wiki/SC1003

# As it is:

"If you want your single-quoted string to end in a backslash, you can
rewrite as 'string'\\ or ignore this warning."

# Possible edit:

"If you want for your single-quoted string to end in a backslash you can
rewrite it as 'string'\\, or if you're quoting a single backslash such as
in `tr -d '\\'`, you can ignore <https://www.shellcheck.net/wiki/ignore> this
warning."

**Comment by brother** [comment]
@wileyhy a edit to the wiki is helpful indeed. but the case pointed out by OP is still valid I think. 

Not sure it will be possible to handle all the permutations with a nice and easy information message though.
Decipher the actual intent is always hard. Did they forget the end single quote? And why was there two back slashes to begin with...

**Comment by brother** [comment]
It was meant as a point in an imaginary internal reasoning leading up to that it's hard to understand the exact intent of code.
If they are missing a backslash, or added one too many. Or was missing a single quote. Or something completely different. People are hard.
